rllib/policy/sample_batch.py

```diff
@@ -352,36 +352,34 @@ def split_by_episode(self) -> List["SampleBatch"]:
             [{"a": [1, 2], "eps_id": [0, 0]}, {"a": [3], "eps_id": [1]}]
         """
 
-        # No eps_id in data -> Make sure there are no "dones" in the middle
-        # and add eps_id automatically.
-        if SampleBatch.EPS_ID not in self:
-            # TODO: (sven) Shouldn't we rather split by DONEs then and not
-            #  add fake eps-ids (0s) at all?
-            if SampleBatch.DONES in self:
-                assert not any(self[SampleBatch.DONES][:-1])
-            self[SampleBatch.EPS_ID] = np.repeat(0, self.count)
+        slices = []
+        if SampleBatch.EPS_ID in self:
+            # Produce a new slice whenever we find a new episode ID.
+            cur_eps_id = self[SampleBatch.EPS_ID][0]
+            offset = 0
+            for i in range(self.count):
+                next_eps_id = self[SampleBatch.EPS_ID][i]
```

Comment:
Wait, if cur_eps_id points to idx 0 and we start the for-loop with i=0, then I'm always comparing the first element to itself. Should we start the for-loop at idx=1, then?

Reply:
No, we still have to append the first episode to `slices`, which we do inside the loop